### ISSUE TITLE
ci: consider removing Tauri MSI bundling from some CI/CD steps

### ISSUE THREAD

**Issue by ReactorScram** [CloudFire-LLC/cloudfire#3922]
I think CD calls into CI and so it probably builds the MSI twice - completely redundant.

If we remove it from `ci.yml` then CD will run it only once and CI won't run it. Devs can still build locally if needed, as discussed during standup.

Re: https://github.com/firezone/firezone/pull/3905#issuecomment-1976897697

**Comment by ReactorScram** [comment]
I'll open a PR to consolidate the unit tests and smoke test for Windows in CI. They both run on both Windows versions, so there's no need to toss out the cache between runs. Almost all of the build time can probably be saved for unit tests if we're going to link the whole Tauri app anyway.

Looking at this CI run: https://github.com/firezone/firezone/actions/runs/8190823437/usage
And this CD run: https://github.com/firezone/firezone/actions/runs/8190495728/usage

And these numbers from a Linux aarch64 VM (building gui-client as a stand-in for amd64 Windows):
- Clean and unit test - 1m12s
- Clean and build - 1m17s
- Clean, unit test, build - 1m11s + 0m23s

**Comment by ReactorScram** [comment]
That didn't work.

So looking at the flowcharts, I think we're actually bottlenecked on the Client image taking 9.5 minutes to build. If we can reduce that, all the integration tests can kick off. The Windows GUI Client actually finishes before the integration tests.

<img width="806" alt="image" src="https://github.com/firezone/firezone/assets/13400041/72d98659-d385-41b7-82fb-83c6992b9bc4">

**Comment by ReactorScram** [comment]
The MSI is a little slow but it's not the bottleneck for CI right now
Closing in favor of #3995
